src/Telemetry/AzureDevOpsClient/AzureDevOpsClient.cs

```diff
@@ -155,6 +162,60 @@ public async Task<WorkItem> CreateRcaWorkItem(string project, string title, Canc
             return JsonConvert.DeserializeObject<WorkItem>(json);
         }
 
+        public delegate bool TryParseLine(string line, out string parsedValue);
+
+        public async Task<HttpResponseMessage> TryGetLogContents(
+            string logUri,
+            CancellationToken cancellationToken)
+        {
+            for (int i = 0; i < _retryNumber; i++)
+            {
+                using (var request = new HttpRequestMessage(HttpMethod.Get, logUri))
+                {
+                    request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("text/plain"));
+
+                    try
+                    {
+                        var response = await _httpClient.SendAsync(request, cancellationToken);
+
+                        if (response.StatusCode == HttpStatusCode.TooManyRequests)
+                        {
+                            if (response.Headers.TryGetValues("Retry-After", out var values) &&
+                                int.TryParse(values.First(), out var seconds))
+                            {
+                                int randomDelay;
+                                lock (_randomNumberGenerator)
+                                {
+                                    randomDelay = _randomNumberGenerator.Next(_retryDelayMin, _retryDelayMax);
+                                }
```

Comment:
This part is repeated twice. Can you just add a private property that does this?

```csharp
private int RandomDelay => _randomNumberGenerator.Next(_retryDelayMin, _retryDelayMax);
```

Feel free to hardcode the numbers there and avoid the two min/max fields too

Reply:
done